src/Castle.Windsor.Extensions.DependencyInjection/Scope/ExtensionContainerScope.cs

```diff
@@ -20,36 +20,40 @@ namespace Castle.Windsor.Extensions.DependencyInjection.Scope
 	using Castle.Core;
 	using Castle.MicroKernel;
 	using Castle.MicroKernel.Lifestyle.Scoped;
-	
+
 	internal class ExtensionContainerScope : ILifetimeScope, IDisposable
 	{
 		public static ExtensionContainerScope Current => current.Value;
 		public static string TransientMarker = "Transient";
 		protected static readonly AsyncLocal<ExtensionContainerScope> current = new AsyncLocal<ExtensionContainerScope>();
 		private readonly ExtensionContainerScope parent;
+		private readonly ExtensionContainerRootScope rootScope;
 		private readonly IScopeCache scopeCache;
 
-		protected ExtensionContainerScope(ExtensionContainerScope parent)
+		protected ExtensionContainerScope(
+			ExtensionContainerScope parent,
+			ExtensionContainerRootScope rootScope)
 		{
 			scopeCache = new ScopeCache();
-			if(parent == null)
-			{
-				this.parent = ExtensionContainerRootScope.RootScope;
-			}
-			else
-			{
-				this.parent = parent;
-			}
+			this.parent = parent ?? rootScope;
+			this.rootScope = rootScope;
 		}
 
-		public static ExtensionContainerScope BeginScope(ExtensionContainerScope parent)
+		public ExtensionContainerRootScope RootScope
+			=> this is ExtensionContainerRootScope myRootScope
```

Comment:
is "myRootScope" a good name?

Do to Windsor code style you might need to use a traditional getter with if statements

Reply:
I agree. I changed it to self, I hope that's better.

Reply:
I changed it to your last suggestion, is better.